src/common/pages/profile-functional.tsx

```diff
@@ -251,16 +296,42 @@ export const Profile = (props: Props) => {
     }
   };
 
-  const bottomReached = () => {
-    const { global, entries, fetchEntries } = props;
+  const bottomReached = async () => {
+    const { global, entries, fetchEntries, match } = props;
+    const { username, section } = match.params;
+
     const { filter, tag } = global;
     const groupKey = makeGroupKey(filter, tag);
 
     const data = entries[groupKey];
     const { loading, hasMore } = data;
 
-    if (!loading && hasMore) {
-      fetchEntries(filter, tag, true);
+    if (section === "trail") {
+      if (dataTrail.entries.length > 0) {
+        setDataTrail({ ...dataTrail, loading: true, fetchedOnce: true });
+
+        let data = await getAccountVotesTrail(
+          username.replace("@", ""),
+          dataTrail.entries[dataTrail.entries.length - 1]!.num! - 1
+        );
+        const sevenDaysAgo = 7 * 24 * 60 * 60 * 1000;
+        while (
+          data.length < 20 &&
+          (new Date().getTime() - new Date(data[0].created!).getTime(), sevenDaysAgo)
+        ) {
+          const moreData = await getAccountVotesTrail(
+            username.replace("@", ""),
+            data[data.length - 1].num! - 1
+          );
+          data = [...moreData, ...data];
+        }
+        let newDataTrail = _.unionBy(dataTrail.entries, data.reverse(), (obj) => obj.post_id);
+        setDataTrail({ ...dataTrail, entries: newDataTrail, loading: false, fetchedOnce: true });
+      }
```

Comment:
Lines 116 - 136 and 311 - 330 seems to perform almost same tasks. To reduce code duplication and improve readability we can move these lines to one function with some modification to use the function in both cases. Also avoid using generic names like obj